src/utils.jl

```diff
@@ -225,41 +229,42 @@ function var_from_nested_derivative(x,i=0)
     end
 end
 
-degree(p::Union{Term,Sym}, sym=nothing) = sym === nothing ? 1 : Int(isequal(p, sym))
-degree(p::Add, sym=nothing) = maximum(degree.(keys(p.dict), sym))
-degree(p::Mul, sym=nothing) = sum(degree(k, sym) * v for (k, v) in p.dict)
-degree(p::Pow, sym=nothing) = p.exp * degree(p.base, sym)
-
 """
     degree(p, sym=nothing)
 
 Extract the degree of `p` with respect to `sym`.
 """
 function degree(p, sym=nothing)
-    p, sym = value(p), value(sym)
-    p isa Number && return 0
-    isequal(p, sym) && return 1
-    p isa Symbolic && return degree(p, sym)
-    throw(DomainError(p, "Datatype $(typeof(p)) not accepted."))
-end
-
-function degree(p::SymbolicUtils.Div, sym=nothing)
-    return degree(p.num,sym)
-end
-
-coeff(p::Union{Term,Sym}, sym=nothing) = sym === nothing ? 0 : Int(isequal(p, sym))
-coeff(p::Pow, sym=nothing) = sym === nothing ? 0 : Int(isequal(p, sym))
-function coeff(p::Add, sym=nothing)
-    if sym === nothing
-        p.coeff
-    else
-        sum(coeff(k, sym) * v for (k, v) in p.dict)
+    p = value(p)
+    sym = value(sym)
+    if p isa Number
+        return 0
     end
-end
-function coeff(p::Mul, sym=nothing)
-    args = unsorted_arguments(p)
-    I = findall(a -> !isequal(a, sym), args)
-    length(I) == length(args) ? 0 : prod(args[I])
+    if isequal(p, sym)
+        return 1
+    end
+    if isterm(p)
+        if sym === nothing
+            return 1
+        else
+            return Int(isequal(p, sym))
+        end
+    elseif ismul(p)
+        return sum(degree(k^v, sym) for (k, v) in zip(keys(p.dict), values(p.dict)))
```

Comment:
```suggestion
        return sum(degree(k, sym) * v for (k, v) in p.dict)
```

In the master branch, `degree` for `Mul` is implemented as 
https://github.com/JuliaSymbolics/Symbolics.jl/blob/0026df9dfa4a97dec7f578c1cae28f58445f2f71/src/utils.jl#L230
to avoid the construction of `k^v`.